Yoast/wordpress-seo
# P3-55 Removed references to React in favor of @wordpress/element

## Context


* See #15535: 

> Currently we have a mix of react imports and @wordpress/element imports. In documentation we refer to React elements as wp.Element, React.Element, ReactElement, React.Component etc. This PR makes sure we consistently use @wordpress/element and every component is documented with either wp.Element or wp.Component for component functions.


## Summary


This PR can be summarized in the following changelog entry:

* Refactor: removes all React references in SEMrush components in favor of @wordpress/element

## Relevant technical choices:

* see summary

## Test instructions

This PR can be tested by following these steps:

* build the JS with `grunt build:js`
* Edit a post
* Enter some focus keyphrase in the Yoast sidebar or in the Yoast metabox
* Click the button to open the modal dialog
* Observe all of the components are present, except the "max related keyphrases" alert
* edit `js/src/components/RelatedKeyPhrasesModal.js` 
* at line 139, change the default value of `maxRelatedKeyphrasesEntered` from `false` to `true`  
* build the JS with `grunt build:js`
* Edit a post
* Enter some focus keyphrase in the Yoast sidebar or in the Yoast metabox
* Click the button to open the modal dialog
* Observe the "max related keyphrases" alert is now present

## UI changes

## Documentation

## Quality assurance


Fixes [P3-55]
